tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,778 @@
+    # Put all portchannel members into dst_ports
+    dst_port_indices = []
+    for _, v in mg_facts['minigraph_portchannels'].items():
+        for member in v['members']:
+            dst_port_indices.append(mg_facts['minigraph_ptf_indices'][member])
+
+    # Generate destination IP's for scale test
+    scale_dest_ips = {}
+    for i in range(1,75):
+        ipv4_rule_name = "FORWARD_RULE_" + str(i)
+        ipv6_rule_name = "V6_FORWARD_RULE_" + str(i)
+        ipv4_address = DST_IP_FORWARDED_scale_PREFIX + str(i)
+        ipv6_address = DST_IPV6_FORWARDED_scale_PREFIX + str(i) + "::1"
+        scale_dest_ips[ipv4_rule_name] = ipv4_address
+        scale_dest_ips[ipv6_rule_name] = ipv6_address
+
+    setup_information = {
+        "blocked_src_port_name" : block_src_port,
+        "blocked_src_port_indice" : block_src_port_indice,
+        "unblocked_src_port_indice" : unblocked_src_port_indice,
+        "scale_port_names" : scale_ports,
+        "scale_port_indices" : scale_ports_indices,
+        "scale_dest_ips" : scale_dest_ips,
+        "dst_port_indices" : dst_port_indices,
+        "router_mac" : router_mac,
+        "bind_ports" : list_ports,
+    }
+
+    return setup_information
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for acl config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
+
+def verify_expected_packet_behavior(exp_pkt, ptfadapter, setup, expect_drop):
+    """Verify that a packet was either dropped or forwarded"""
+    if expect_drop:
+        testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["dst_port_indices"])
+    else:
+        testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=setup["dst_port_indices"],
+                                        timeout=20)
+
+def generate_forward_packets(setup, dst_ip = DST_IP_FORWARDED_ORIGINAL, dst_ipv6 = DST_IPV6_FORWARDED_ORIGINAL):
+    """Generate packets that match the destination IP of given ips.
+    If no IP is given, default to our original forwarding ips"""
+
+    forward_packets = {}
+
+    forward_packets["IPV4"] = testutils.simple_tcp_packet(eth_dst=setup["router_mac"],
+                                ip_src=IP_SOURCE,
+                                ip_dst=dst_ip)
+
+    forward_packets["IPV6"] = testutils.simple_tcpv6_packet(eth_dst=setup["router_mac"],
+                                ipv6_src=IPV6_SOURCE,
+                                ipv6_dst=dst_ipv6)
+
+    return forward_packets
+
+
+def generate_drop_packets(setup):
+    """Generate packets with a DST_IP that is never included in our forward rules"""
```

Comment:
`generate_forward_packets` and `generate_drop_packets` can be implemented as one single function, the only difference is the dst_ip.